bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,255 @@
+        """
+        func_args = self.func_args
+        for p in self.params:
+            if p in func_args: # the parameter value has been given in func_args
+                value_of_p = func_args[p]
+                type_of_p = self.find_type(value_of_p)
+                if type(value_of_p) == bool: # Bool(s) in python are represented as True/False but in C++ are true/false. The following if statements are to account for that
+                    if value_of_p: value_of_p = 'true'
+                    else: value_of_p = 'false'
+            else: # the parameter was not in func_args. Thus this parameter has to be mapped to a column of rdf
+                if p not in self.col_names:
+                    raise Exception(f"Unable to map function argument {p} to a column.\nUse correct name of column or pass a list of column names.")
+                value_of_p = p 
+                type_of_p = self.find_type(p)
+            self.args_info[p] = (type_of_p, value_of_p)
+        return self
+
+    def generate_function_call(self):
+        """
+        Generates a function call and signature. Gives a unique name to the function if it is a lambda function.
+        Updates the class with new attributes func_call and func_sign to hold them,
+        
+        Returns:
+            self: An instance of the class
+        """
+        func = self.func
+        if func.__name__ == '<lambda>' : 
+            func.__name__ = f"_lambda_func_number_{FunctionJitter.lambda_function_counter}"
+            FunctionJitter.lambda_function_counter += 1
+        self.func_call = f"{func.__name__}({', '.join(str(arg_info[1]) for arg_info in self.args_info.values())})"
+        self.func_sign = [str(self.args_info[p][0]) for p in self.params]
+        return self
+            
+    def jit_function(self, func, cols_list, extra_args):
+        """
+        Jits the provided function using ROOT's NumbaDeclare.
+        Also checks if the function was jitted earlier in which case it won't jit again but if signature does not match. It raises an error.
+
+        Arguments:
+        func: A python callable
+        cols_list: A list of columns of RDF on which func depends on.
+        extra_args: A dict of extra arguments that func requires.
+        """
+        if func.__name__ in FunctionJitter.function_cache:
+            func_call, func_sign = FunctionJitter.function_cache[func.__name__]
+            self.find_function_params(func)
+            self.generate_func_args(cols_list, extra_args)
+            self.find_function_signature()
+            self.generate_function_call()
+            if self.func_sign != func_sign:
+                raise ValueError("Trying to re-use a function. Do not change function signature.".format(func))
+            return self.func_call
+        self.find_function_params(func)
+        self.generate_func_args(cols_list, extra_args)
+        self.find_function_signature()
+        self.generate_function_call()
```

Comment:
In all these function calls you're never using the return value, which is `self`, you should remove those return statements.

Then, a second aspect of this is that each function will modify the state of the class by dynamically injecting some new attribute. For example the method `find_function_params` defines `self.func`, `self.params`, `self.args_info`. It might be easier to read if each function instead returned those values, without side effects. This improvement can go in another PR though